src/CAN/can_mapping.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "can_mapping.h"
+#include "byteswap.h"
+#include <byteswap.h>
+
+#include <stdio.h>
+
+float canmapping_extract_value(uint64_t raw_data, CANMapping *mapping)
+{
+	uint8_t offset = mapping->offset;
+	uint8_t length = mapping->length;
+	if (! mapping->bit_mode) {
+			length *= 8;
+			offset *= 8;
+	}
+	uint32_t bitmask = (1UL << length) - 1;
+	uint32_t raw_value = (raw_data >> offset) & bitmask;
+	if (mapping->big_endian) {
+	        switch (mapping->length) {
+	            case 2:
+	                raw_value = swap_uint16(raw_value);
+	                break;
+	            case 3:
+	                raw_value = swap_uint24(raw_value);
+	                break;
+	            case 4:
+	                raw_value = swap_uint32(raw_value);
+	                break;
+	            default:
+	                break;
+	        }
+	}
+	return (float)raw_value;
+}
+
+float canmapping_apply_formula(float value, CANMapping *mapping)
+{
+		value *= mapping->multiplier;
+		if (mapping->divider)
```

Comment:
I agree with you on principal. However, industry standard CAN mapping schemes provides for a somewhat redundant divider. Explained this in the last PR, you might've missed it.   Overall, CAN mapping standards includes a divider, it's an industry thing. 

I didn't have it initially, and I initially resisted including it, but after reading and seeing different CAN mapping schemes for devices out there, I included it after allo.